mmpose/datasets/datasets/body3d/body3d_mpi_inf_3dhp_dataset.py

```diff
@@ -0,0 +1,385 @@
+
+    `Monocular 3D Human Pose Estimation In The Wild Using Improved CNN
+    Supervision` 3DV'2017
+    More details can be found in the `paper
+    <https://arxiv.org/pdf/1611.09813>`__.
+
+    MPI-INF-3DHP keypoint indexes:
+        0: 'head_top',
+        1: 'neck',
+        2: 'right_shoulder',
+        3: 'right_elbow',
+        4: 'right_wrist',
+        5: 'left_shoulder;,
+        6: 'left_elbow',
+        7: 'left_wrist',
+        8: 'right_hip',
+        9: 'right_knee',
+        10: 'right_ankle',
+        11: 'left_hip',
+        12: 'left_knee',
+        13: 'left_ankle',
+        14: 'root (pelvis)',
+        15: 'spine',
+        16: 'head'
+
+    Args:
+        ann_file (str): Path to the annotation file.
+        img_prefix (str): Path to a directory where images are held.
+            Default: None.
+        data_cfg (dict): Data configurations. Please refer to the docstring of
+            Body3DBaseDataset for common data attributes. Here are MPI-INF-3DHP
+            specific attributes.
+            - joint_2d_src: 2D joint source. Options include:
+                "gt": from the annotation file
+                "detection": from a detection result file of 2D keypoint
+                "pipeline": will be generate by the pipeline
+                Default: "gt".
+            - joint_2d_det_file: Path to the detection result file of 2D
+                keypoint. Only used when joint_2d_src == "detection".
+            - need_camera_param: Whether need camera parameters or not.
+                Default: False.
+        pipeline (list[dict | callable]): A sequence of data transforms.
+        test_mode (bool): Store True when building test or
+            validation dataset. Default: False.
+    """
+
+    JOINT_NAMES = [
+        'HeadTop', 'Neck', 'RShoulder', 'RElbow', 'RWrist', 'LShoulder',
+        'LElbow', 'LWrist', 'RHip', 'RKnee', 'RAnkle', 'LHip', 'LKnee',
+        'LAnkle', 'Root', 'Spine', 'Head'
+    ]
+
+    # 2D joint source options:
+    # "gt": from the annotation file
+    # "detection": from a detection result file of 2D keypoint
+    # "pipeline": will be generate by the pipeline
+    SUPPORTED_JOINT_2D_SRC = {'gt', 'detection', 'pipeline'}
+
+    # metric
+    ALLOWED_METRICS = {
+        'mpjpe', 'p-mpjpe', '3dpck', 'p-3dpck', '3dauc', 'p-3dauc'
+    }
+
+    def load_config(self, data_cfg):
+        super().load_config(data_cfg)
+        # mpi-inf-3dhp specific attributes
+        self.joint_2d_src = data_cfg.get('joint_2d_src', 'gt')
+        if self.joint_2d_src not in self.SUPPORTED_JOINT_2D_SRC:
+            raise ValueError(
+                f'Unsupported joint_2d_src "{self.joint_2d_src}". '
+                f'Supported options are {self.SUPPORTED_JOINT_2D_SRC}')
+
+        self.joint_2d_det_file = data_cfg.get('joint_2d_det_file', None)
+
+        self.need_camera_param = data_cfg.get('need_camera_param', False)
+        if self.need_camera_param:
```

Comment:
Is it allowed for MPI-INF-3DHP to not have camera parameters? If not, consider adding an assertion here.